src/devices/oregon_scientific.c

```diff
@@ -50,23 +50,25 @@ static float get_os_temperature(unsigned char *message)
 {
     float temp_c = 0;
     temp_c = (((message[5] >> 4) * 100) + ((message[4] & 0x0f) * 10) + ((message[4] >> 4) & 0x0f)) / 10.0F;
-    if (message[5] & 0x0f)
+    // Correct 0x0f to 0x08:
+    if (message[5] & 0x08) {
         temp_c = -temp_c;
+    }
     return temp_c;
 }
 
 static float get_os_rain_rate(unsigned char *message)
 {
-    float rain_rate = 0; // Nibbles 11..8 rain rate, LSD = 0.01 inches per hour
-    rain_rate = (((message[5] & 0x0f) * 1000) + ((message[5] >> 4) * 100) + ((message[4] & 0x0f) * 10) + ((message[4] >> 4) & 0x0f)) / 100.0F;
+    float rain_rate = 0.0F; // Nibbles 11..8 rain rate, LSD = 0.1 units per hour, 4321 = 123.4 units per hour
```

Comment:
This is later used as "rain_rate_in_h", can you confirm that units is in/h here? Or Is a unit 0.1 in, thus the change of scale wrong here?

Reply:
Hi, good question.  Not sure why I edited that section...  The sensor generated rain_rate lags so much I don't use it myself (I calculate it based on when the rain_total field changes.)  I used the page here: https://www.bashewa.com/wmr200-protocol.php as a starting point and compared it with the data I get from a PCR800 sensor.  I believe the PCR800 reports it in units of 0.1 inches per hour.